File: src/main/kotlin/at/searles/parsingtools/properties/PropertiesInitializer.java
package at.searles.parsingtools.properties;

import at.searles.parsing.Initializer;
import at.searles.parsing.ParserStream;

public class PropertiesInitializer implements Initializer<Properties> {

    private static class Holder {
        static Properties instance = new Properties();
    }

    @Override
    public Properties parse(ParserStream stream) {
        return Holder.instance;
    }

    @Override
    public boolean consume(Properties properties) {
        return properties.isEmpty();
    }

    @Override
    public String toString() {
        return "{empty properties}";
    }
}

File: src/main/kotlin/at/searles/parsingtools/properties/PropertiesInitializer.kt
package at.searles.parsingtools.properties

import at.searles.parsing.Initializer
import at.searles.parsing.ParserStream

class PropertiesInitializer : Initializer<Properties> {

    private object Holder {
        internal var instance = Properties()
    }

    override fun parse(stream: ParserStream): Properties? {
        return Holder.instance
    }

    override fun consume(properties: Properties): Boolean {
        return properties.isEmpty()
    }

    override fun toString(): String {
        return "{empty properties}"
    }
}